whois-api/src/main/java/net/ripe/db/whois/api/fulltextsearch/FulltextSearch.java

```diff
@@ -54,6 +57,29 @@ protected SearchResponse.Lst createVersion() {
         return result;
     }
 
+    protected String customizeIfFiltering(final String value){
+        final Matcher matcher = Pattern.compile("\\((.*?\\))").matcher(value);
+        if (!matcher.find()) {
+            return value;
+        }
+        final String fullAttributeValue = matcher.group(1);
+        final String attributeType = fullAttributeValue.replace("(", "").split(":")[0];
```

Comment:
I think you are assuming too many things here. query can have multiple attribute, multiple AND OR operators. This might work for single attribute value. We need to look at first some setting or mappings in ES rather than manually modifying query like this

Reply:
See the test cases. There is test cases for multiple attributes, or, and...multiple object types...etc
This is working for multiple values as well. (I take all this examples from db-web-ui calls)

I tried to find some settings or mappings but the bug is in the query...we should use all the filters that we are specifying, not just one